packages/knip/src/plugins/karma/index.ts

```diff
@@ -0,0 +1,101 @@
+import type { IsPluginEnabled, Plugin, ResolveConfig, ResolveEntryPaths } from '../../types/config.js';
+import { type Input, toDevDependency, toEntry } from '../../util/input.js';
+import { extname, isInternal, join } from '../../util/path.js';
+import { hasDependency } from '../../util/plugin.js';
+import type { Config, ConfigOptions } from './types.js';
+
+// https://karma-runner.github.io/latest/config/configuration-file.html
+
+const title = 'Karma';
+
+const enablers = ['karma'];
+
+const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependencies, enablers);
+
+//👇 All but CoffeeScript ones. Low usage nowadays compared to the effort to implement support for those files
+const config = ['karma.conf.js', 'karma.conf.ts', '.config/karma.conf.js', '.config/karma.conf.ts'];
+
+const entry: string[] = [];
+
+type ConfigFile = (config: Config) => void;
+
+const resolveConfig: ResolveConfig<ConfigFile> = async (configFile, options) => {
+  const inputs = new Set<Input>();
+
+  const config = loadConfigFromFile(configFile);
+
+  if (config.frameworks) {
+    for (const framework of config.frameworks) {
+      inputs.add(toDevDependency(devDepForFramework(framework)));
+    }
+  }
+  if (config.plugins) {
+    for (const plugin of config.plugins) {
+      if (typeof plugin !== 'string') continue;
+      if (isInternal(plugin)) {
+        inputs.add(toEntry(extname(plugin).length === 0 ? `${plugin}.js` : plugin));
```

Comment:
Usually it's better to either:

- if we're sure it's internal and allow it to end up as "unresolved": use `toDeferResolveEntry` - this is more precise/performant, especially combined with `toDevDependency`
- otherwise, use `toDeferResolve` for both internal/external `plugin` strings - this is what most plugins do

Both will defer resolution to later in the process to an internal `enhanced-resolve` wrapper, taking care of file extensions and other module resolution gotchas.

Reply:
Thanks! Went for `toDeferResolveEntry` / `toDevDependency`. And removed the `extname` trick thingie. I came across those but didn't fully understand them so that's why I used the basic ones. Resolving.